### PR TITLE
LPS-204513 Loading screen when user should not see any content

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPS-204513](https://liferay.atlassian.net/browse/LPS-204513)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by p-albuquerque** [comment]
I'm aware that permission verifications use to be done in _Service_ file. After talk with Pedro, I understood that the guidance to that case was perform this treatment  inside _LocalService_ through DSLQuery (as `ObjectEntryLocalServiceImpl`). Given that, I have no questions/suggestions to do.

**Review by p-albuquerque — APPROVED** [review]
I'm aware that permission verifications use to be done in _Service_ file. After talk with Pedro, I understood that the guidance to that case was perform this treatment inside _LocalService_ through DSLQuery (as `ObjectEntryLocalServiceImpl`). Given that, I have no questions/suggestions to do.

**Review by p-albuquerque — CHANGES_REQUESTED** [review]
A few points about the integration test

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java [hunk 1]
```diff
@@ -264,6 +279,51 @@ public void testDeleteObjectFolderItemByObjectFolderId() {
...
+			_resourcePermissionLocalService.setResourcePermissions(
+				TestPropsValues.getCompanyId(), ObjectFolder.class.getName(),
+				ResourceConstants.SCOPE_COMPANY,
+				String.valueOf(TestPropsValues.getCompanyId()),
+				role.getRoleId(), new String[] {ActionKeys.VIEW});
+
+			_userLocalService.addRoleUser(role.getRoleId(), user);
+
+			ObjectFolder objectFolder =
+				_objectFolderLocalService.getUncategorizedObjectFolder(
```
Comment (p-albuquerque) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java:306]: Do uncategorized ObjectFolder has some ObjectItens? I don't see any ObjectDefinition linked to this folder either in this test and in set up.

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java [hunk 2]
```diff
@@ -264,6 +279,51 @@ public void testDeleteObjectFolderItemByObjectFolderId() {
...
+					TestPropsValues.getCompanyId());
+
+			List<ObjectFolderItem> objectFolderItems =
+				_objectFolderItemLocalService.
+					getObjectFolderItemsByObjectFolderId(
+						objectFolder.getObjectFolderId());
+
+			Assert.assertEquals(
+				objectFolderItems.toString(), 0, objectFolderItems.size());
+
```
Comment (p-albuquerque) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java:316]: Maybe you can add a new scenario in this test:
Once this ObjectFolder has some ObjectDefinition linked. Give view permission over this ObjectDefinition and assert that the user can _see_ it

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java:306]: System objectDefinitions are into uncategorized folder

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFolderItemLocalServiceImpl.java [hunk 3]
```diff
@@ -151,7 +158,37 @@ public List<ObjectFolderItem> getObjectFolderItemsByObjectDefinitionId(
...
+
+				PermissionChecker permissionChecker =
+					PermissionThreadLocal.getPermissionChecker();
+
+				if (permissionChecker == null) {
+					return predicate;
+				}
+
+				Predicate permissionPredicate =
+					_inlineSQLHelper.getPermissionWherePredicate(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFolderItemLocalServiceImpl.java:179]: Hey @pedro-oliveira446 , what happens if _SQL Check Enabled_ is turned off? Maybe you should include some piece of code like [this one](https://github.com/liferay/liferay-portal/blob/2b6e533dba0b42c5c215935188f03065a3ce744e/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L2801-L2806).
You can configure this on "**System Settings > Security Tools > SQL Check Enabled**"

Comment (guilhermedcamacho) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFolderItemLocalServiceImpl.java:179]: Another thing, should we pass the `groupId` when calling `_inlineSQLHelper.getPermissionWherePredicate`? 
You can ask Gabriel if this makes any difference in this context.

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
@guilhermedcamacho 
NOTE: I'm now in charge of this issue, this decision was made in agreement with Samuel. I will open a new PR soon, after deal with the requested actions.

**Comment by p-albuquerque** [comment]
Resent: https://github.com/liferay-objects/liferay-portal/pull/3024

**Pull request closed without merging**